public/app/plugins/panel/geomap/editor/StyleRuleEditor.tsx

```diff
@@ -1,31 +1,60 @@
-import React, { ChangeEvent, FC, useCallback } from 'react';
+import React, { FC, useCallback, useMemo } from 'react';
 import { GrafanaTheme2, SelectableValue, StandardEditorProps } from '@grafana/data';
 import { ComparisonOperation, FeatureStyleConfig } from '../types';
-import { Button, ColorPicker, InlineField, InlineFieldRow, Input, Select, useStyles2 } from '@grafana/ui';
+import { Button, ColorPicker, InlineField, InlineFieldRow, Select, useStyles2 } from '@grafana/ui';
 import { css } from '@emotion/css';
 import { NumberInput } from 'app/features/dimensions/editors/NumberInput';
+import { Observable } from 'rxjs';
+import { useObservable } from 'react-use';
+import { FeatureLike } from 'ol/Feature';
 
 export interface StyleRuleEditorSettings {
   options: SelectableValue[];
+  features: Observable<FeatureLike[]>;
+  properties: Observable<SelectableValue[]>;
 }
 
 export const StyleRuleEditor: FC<StandardEditorProps<FeatureStyleConfig, any, any, StyleRuleEditorSettings>> = (
   props
 ) => {
   const { value, onChange, item } = props;
+
   const settings: StyleRuleEditorSettings = item.settings;
+  const { options, features, properties } = settings;
+
+  const propertyOptions = useObservable(properties);
+  const feats = useObservable(features);
+
+  const uniqueValues = useMemo(() => {
+    const uniqueValues: SelectableValue[] = [];
+    if (feats) {
+      if (value?.rule) {
+        const values = [];
+        for (let f = 0; f < feats.length; f++) {
+          if (value.rule) {
+            values.push(feats[f].get(value.rule.property));
+          }
+        }
```

Comment:
We could potentially simplify this logic in the following way:

```suggestion
  const uniqueValues = useMemo(() => {
    const generatedUniqueValues: SelectableValue[] = [];
    if (feats) {
      if (value?.rule) {
        for (let f = 0; f < feats.length; f++) {
          const featureValue = feats[f].get(value.rule.property)
          const newUniqueValue: SelectableValue = { value: featureValue, label: `${featureValue}` }
          
          if (!generatedUniqueValues.some(v => v.value === newUniqueValue.value)) {
            generatedUniqueValues.push(newUniqueValue)  
          }
        }
      }
    }

    return generatedUniqueValues.sort();
  }, [feats, value]);
```